src/Parallel/DataCollector.h

```diff
@@ -0,0 +1,86 @@
+#ifndef SEISSOL_PARALLEL_DATACOLLECTOR_H
+#define SEISSOL_PARALLEL_DATACOLLECTOR_H
+
+#include <cstddef>
+#include <vector>
+
+#include "Kernels/precision.hpp"
+
+#ifdef ACL_DEVICE
+#include "device.h"
+#endif
+
+namespace seissol::parallel {
+
+class DataCollector {
```

Comment:
Proposed comment (ChatGPT)
// Class for collecting and managing data, facilitating data transfers between host and device